lib/grant.js

```diff
@@ -285,14 +295,16 @@ function useClientCredentialsGrant (done) {
   }
 
   var self = this;
-  return this.model.getUserFromClient(clientId, clientSecret,
-      function (err, user) {
+  return this.model.getUserFromClient(clientId, clientSecret, function (err, user) {
     if (err) return done(error('server_error', false, err));
+
     if (!user) {
       return done(error('invalid_grant', 'Client credentials are invalid'));
     }
 
     self.user = user;
+    self.scope = self.client.scope;
```

Comment:
This currently always assigns `undefined`. I see two possibilities here: either the client is allowed to solicit scopes on its own behalf when issuing the client_credentials grant request, or the model's `getClient` will need to include a scope string (as proposed previously by @nunofgs). Is it sound to let the client request its own scope? Any input here?